docs/ProfilingNonJavaApplications.md

```diff
@@ -0,0 +1,36 @@
+# Profiling Non-Java Applications
+
+The scope of profiling non-java applications is limited to the case when profiler is controlled
+programmatically from the process being profiled. Similar to the
+[Java API](https://github.com/async-profiler/async-profiler/blob/master/docs/OtherUseCases.md#using-java-api),
+there is a C API for using inside native applications.
+```
+typedef const char* asprof_error_t;
+typedef void (*asprof_writer_t)(const char* buf, size_t size);
+
+// Should be called once prior to any other API functions
+DLLEXPORT void asprof_init();
+typedef void (*asprof_init_t)();
+
+// Returns an error message for the given error code or NULL if there is no error
+DLLEXPORT const char* asprof_error_str(asprof_error_t err);
+typedef const char* (*asprof_error_str_t)(asprof_error_t err);
+
+// Executes async-profiler command using output_callback as an optional sink
+// for the profiler output. Returning an error code or NULL on success.
```

Comment:
Worth mentioning what the format of the command is.

Reply:
Addressed with an example while addressing the below comment